### ISSUE TITLE
Sharpshooters have a default pistol stuck in their secondary weapon slot, can't equip a better pistol.

### ISSUE THREAD

**Issue by chloeemcarthur** [long-war-2/lwotc#850]
My sharpshooter has a conventional pistol in her secondary weapon slot, which does not show better pistols such as the mag pistol as an option to replace it with. The utility slot does show better pistols such as the mag pistol but they are unable to be selected because of a message saying: "Max 1 unknown weapon category per soldier".

Any ideas what causes this?

**Comment by Grobobobo** [comment]
Likely a config screw-up/mod conflict, sharpshooters should not have a pistol as a secondary in the first place.

**Comment by ScaryPunkGhost** [comment]
I can't think of anything that would be causing a conflict, here's my modlist, excluding cosmetic mods:
Mod Config Menu
Unrestricted Customization
Full Character Customization from Start
Load from character pool
Use my Class
Revert Overwatch Rules Change
Better Second Wave Mod Support
WOTC Flawless
Numeric Health Display
Evac all
Robojumpers Squad Select

**Comment by Grobobobo** [comment]
Use my class may have forced a soldier to become a vanilla sharpshooter instead of the lwotc one

**Comment by ScaryPunkGhost** [comment]
I remade the characters affected and this solved it, so that appeared to be the issue, however now my sharpshooters pistol abilities (specifically lightning hands) say they have insufficient ammo, and I have multiple copies of the ability in my hotbar.

**Comment by Grobobobo** [comment]
The lightning hands might have been bound to the secondary slot instead of the pistol slot because of it. That said, rebuilding the soldier class with console commands should fix that issue

**Comment by ScaryPunkGhost** [comment]
How would i go about doing that? What commands do I use?

**Comment by Grobobobo** [comment]
https://steamcommunity.com/sharedfiles/filedetails/?id=1370543410

**Comment by ScaryPunkGhost** [comment]
So I should just use the rebuild selected soldiers class command? Do I need to specify to rebuild them as a sharpshooter and if so is there a unique name for the long war version of the sharpshooter class?

**Comment by Grobobobo** [comment]
Yes, It's LWS_Sharpshooter or something, check in configs because I'm not sure

**Comment by ScaryPunkGhost** [comment]
Rebuilt my sharpshooters and the problem persists, specced into clutch shot as well and it still says insufficient ammo on the ability.

**Comment by ScaryPunkGhost** [comment]
There's also still the issue of having multiple copies of the ability in my hotbar.

**Comment by Grobobobo** [comment]
Hmm, I don't exactly know what use my class broke then, can you get away with starting a new campaign? Also, by "utility slot" do you mean dedicated pistol slot?

**Comment by ScaryPunkGhost** [comment]
I don't have a dedicated pistol slot, just armor, primary and secondary weapons and three utility slots.

Also no, I can't really start a new campaign.

**Comment by Grobobobo** [comment]
Wait, what? Beta 2 has added a dedicated pistol slot mod as a requiriment. Are you playing an outdated version?

**Comment by ScaryPunkGhost** [comment]
I don't think so, but I don't remember seeing a dedicated pistol mod in the requirements, maybe I missed it. Can you link me to the mod?

**Comment by ScaryPunkGhost** [comment]
Wait, found it. I'll test it and see if it fixes the problem

**Comment by Grobobobo** [comment]
Are you sure that you're playing on beta 2? You should not have access to utility pistols at all anymore.

**Comment by ScaryPunkGhost** [comment]
Maybe not, I installed recently but maybe I installed from the wrong page by accident. Is it possible to update mid playthrough?

**Comment by Grobobobo** [comment]
It is, though i rember a mid-update causing some nee issues on its own, so be careful.

**Comment by ScaryPunkGhost** [comment]
It would appear that this has not entirely solved the problem. Although I now have a sidearm slot, I can only put certain pistols in the slots, such as the autopistol and traditional pistol. The gauss pistol for example can only be equipped in the utility slots.

**Comment by Morflight** [comment]
I have a similar problem so I suppose they are related somehow.

Impacted classes : Reaper, Psi Operative, Skirmisher

All of the mentionned classes have a default pistol equipped that I couldn't use. (I believe it appeared when I purchased a plated version of the skirmishers/reapers armor and when i promoted my psi ops).

Not a gambreaking bug, I just have -1 mobility and can't use the guns in game, I could live with that.

Although recently I just loaded a save and the aforementionned classes (except Skirmishers) all have -10/-11 mobility and now I have 5 pistol overwatch icons on my psi ops (not tested on repaer yet).

Now that's a problem :/

In game I found my psi ops to be severely slowed down (but I don't believe the -10/-11 have taken full effect).

Now that's becoming a problem.

I only installed the required and recommented mods (and for some reason the additionnal icons won't work for me) after a clean XCOM2 reinstall and following the installation instructions.

Any tip ? Maybe I could manually remove equipment from my soldiers. If someone would be kind enough to give me a way to dump my chatacter's current inventory and/or a way to edit it I would be very grateful.
